helm/nginx-ingress-controller-app/templates/controller-deployment.yaml

```diff
@@ -136,15 +136,9 @@ spec:
         {{- if .Values.controller.extraEnvs }}
           {{- toYaml .Values.controller.extraEnvs | nindent 8 }}
         {{- end }}
-        {{- if .Values.cluster.profile }}
-        {{- if gt (.Values.cluster.profile | int) 2 }}
+        {{- if .Values.controller.resources }}
```

Comment:
Not a blocker, but you could experiment with `with` here to avoid the `if`

https://helm.sh/docs/chart_template_guide/control_structures/#modifying-scope-using-with